packages/deployments/contracts/contracts/messaging/RootManager.sol

```diff
@@ -33,14 +33,65 @@ contract RootManager is ProposedOwnable, IRootManager, WatcherClient, DomainInde
 
   event RootDiscarded(bytes32 fraudulentRoot);
 
+  event ProposerAdded(address proposer);
+
+  event ProposerRemoved(address proposer);
+
   event ConnectorAdded(uint32 domain, address connector, uint32[] domains, address[] connectors);
 
   event ConnectorRemoved(uint32 domain, address connector, uint32[] domains, address[] connectors, address caller);
 
   event PropagateFailed(uint32 domain, address connector);
 
+  event SlowModeActivated();
+
+  event OptimisticModeActivated();
+
+  event OptimisticRootPropagated(bytes32 aggregateRoot, bytes32 domainsHash);
+
+  event AggregateRootProposed(
+    uint256 snapshotId,
+    uint256 endOfDispute,
+    bytes32 aggregateRoot,
+    bytes32 baseRoot,
+    bytes32[] snapshotsRoots,
+    uint32[] domains
+  );
+
+  event ProposedRootFinalized(bytes32 aggregateRoot);
+
+  // ============ Errors ============
+
+  error RootManager_InvalidSnapshotId(uint256 snapshotId);
+
+  error RootManager_InvalidDomains();
+
+  error RootManager_InvalidAggregateRoot();
+
+  error RootManager_SlowModeOn();
+
+  error RootManager_OptimisticModeOn();
+
+  error RootManager_ProposeInProgress();
+
+  error RootManager_NotWhitelistedProposer(address caller);
+
+  error RootManager_EmptyFinalizedOptimisticRoot();
+
+  error RootManager_OldAggregateRoot();
+
   // ============ Properties ============
 
+  /**
+   * @notice The time watchers have to detect and invalidate the proposed root.
+   */
+  uint256 public constant DISPUTE_TIME = 30 minutes;
+
+  /**
+   * @notice Duration of the snapshot
+   */
+  uint256 public constant SNAPSHOT_DURATION = 30 minutes;
+
```

Comment:
We talked with @skeletor-spaceman about this... We consider that leaving the SNAPSHOT_DURATION as timestamp is more intuitive and does not represents any danger to the system. Also easier and less risky to be set on every domain since will use the same value on all of them.

About changing the DISPUTE_TIME to be in blocks, we consider that is a good idea so we will implement that.

Reply:
lol I think this is the comment that I thought was "above"

agree with the assessment of both!